crates/operation-checks/src/aggregate_field_usage.rs

```diff
@@ -19,6 +40,64 @@ pub struct FieldUsage {
     pub(crate) type_condition_counts: HashMap<String, u64>,
 }
 
+impl UsageProvider for FieldUsage {
+    fn field_is_used(&self, field_id: schema::FieldId) -> bool {
+        self.count_per_field.contains_key(&field_id)
+    }
+
+    fn argument_is_used(&self, argument_id: schema::ArgumentId) -> bool {
+        self.count_per_field_argument.contains_key(&argument_id)
+    }
+
+    fn enum_value_is_used(&self, enum_and_value: &str) -> bool {
+        self.count_per_enum_value.contains_key(enum_and_value)
+    }
+
+    fn argument_is_left_out(&self, argument_id: schema::ArgumentId) -> bool {
+        self.arguments_with_defaults_left_out_count.contains_key(&argument_id)
+    }
+
+    fn type_condition_is_used(&self, type_condition: &str) -> bool {
+        self.type_condition_counts.contains_key(type_condition)
+    }
+
+    fn used_argument_ids(&self) -> Box<dyn Iterator<Item = schema::ArgumentId> + '_> {
+        Box::new(self.count_per_field_argument.keys().copied())
+    }
+}
+
+/// A usage provider that assumes all fields, arguments, and enum values are used.
+/// This is useful for checking breaking changes without requiring actual operation data.
+#[derive(Debug, Default)]
+pub struct AssumeAllUsed;
+
+impl UsageProvider for AssumeAllUsed {
+    fn field_is_used(&self, _field_id: schema::FieldId) -> bool {
+        true
+    }
+
+    fn argument_is_used(&self, _argument_id: schema::ArgumentId) -> bool {
+        true
+    }
+
+    fn enum_value_is_used(&self, _enum_and_value: &str) -> bool {
+        true
+    }
+
+    fn argument_is_left_out(&self, _argument_id: schema::ArgumentId) -> bool {
+        // When assuming all usage, we also assume arguments with defaults might be left out
+        true
+    }
+
+    fn type_condition_is_used(&self, _type_condition: &str) -> bool {
+        true
+    }
+
+    fn used_argument_ids(&self) -> Box<dyn Iterator<Item = schema::ArgumentId> + '_> {
```

Comment:
Let's see how it works out in practice, but I think we may have to change the signature for this one (is_argument_used()).

Reply:
Ah that's new, I didn't get that at first sight. Checking it out in more depth.

Reply:
I think this will cause breaking changes in input types to be missed, because if the iterator is empty, the input types used in field arguments will not be considered used, and will not trigger breaking change diagnostics as a consequence.

I can see at least two ways to fix that: 

1. An extra `assume_all_input_types_used() -> bool` method on the trait to short-circuit the whole find_used_input_types() implementation
2. Or we move find_used_input_types to the trait